src/kOS/Module/kOSProcessor.cs

```diff
@@ -528,8 +528,15 @@ public void SetMode(ProcessorModes newProcessorMode)
                 switch (newProcessorMode)
                 {
                     case ProcessorModes.READY:
-                        if ((ProcessorMode == ProcessorModes.STARVED || ProcessorMode == ProcessorModes.OFF) && shared.Cpu != null)
-                            shared.Cpu.Boot();
+                        if (Config.Instance.StartOnArchive)
```

Comment:
Unless I'm reading the code wrong, the explanation that @erendrake provides is why I removed it.  We only reach the switch if the ProcessorMode != newProcessorMode, and therefore newProcessorMode must be either STARVED or OFF by definition.  Realistically, the extra check won't have a significant effect on performance, but I just thought I would clear out a redundancy while I was adding the volume switch.